### PR TITLE
Adds currently playing and copy URL commands.

### PR DESCRIPTION
**Pull request by acristoffers**
Adds two more commands to make this extension on pair with the script version: "Current Song" and "Copy Current Song's URL".

### CODE REVIEW

File: extensions/spotify-controls/src/utils.ts [hunk 1]
```diff
@@ -15,3 +15,14 @@ export async function runAppleScriptSilently(appleScript: string) {
...
+/**
+ * Runs the AppleScript
+ *
+ * @param appleScript - The AppleScript to run
+ * @throws An error when the AppleScript fails to run
+ * @returns A promise that is resolved when the AppleScript finished running
+ */
+export async function runAppleScriptAndReturn(appleScript: string) {
+  return await runAppleScript(appleScript);
+}
```
Comment (thomaspaulmann) [inline extensions/spotify-controls/src/utils.ts:28]: Do you think we need this? Seems like the function doesn't add anything on top of the `runAppleScript` one. How about keeping it simple and remove `runAppleScriptAndReturn` and use `runAppleScript` directly?

File: extensions/spotify-controls/src/copyUrl.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    end tell
+
+    set AppleScript's text item delimiters to ":"
+    set idPart to third text item of spotifyURL
+
+    set the clipboard to ("https://open.spotify.com/track/" & idPart)`;
+
+  try {
+    await runAppleScriptAndReturn(script);
+    await showToast(ToastStyle.Success, "Spotify", "URL copied to the clipboard.");
```
Comment (thomaspaulmann) [inline extensions/spotify-controls/src/copyUrl.ts:17]: It's best to show the important information in the title. I would suggest something like this:
```suggestion
    await showToast(ToastStyle.Success, "Copied URL to clipboard");
```

File: extensions/spotify-controls/src/current.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      return currentTrackName & " by " & currentTrackArtist & " (Paused)"
+  else
+      return "Not playing"
+  end if`;
+
+  try {
+    const result = await runAppleScriptAndReturn(script);
+    await showToast(ToastStyle.Success, "Spotify", result);
+  } catch (err) {
+    await showToast(ToastStyle.Failure, "Spotify", "Could not fetch current track.");
```
Comment (thomaspaulmann) [inline extensions/spotify-controls/src/current.ts:34]:
```suggestion
    await showToast(ToastStyle.Failure, "Failed getting playing track");
```

File: extensions/spotify-controls/src/utils.ts (hunk 1 above)
Comment (acristoffers) [reply extensions/spotify-controls/src/utils.ts:28]: I just copied the style I saw without thinking about it. It makes sense to just use the function directly.

Comment (thomaspaulmann) [reply extensions/spotify-controls/src/utils.ts:28]: Want to make this change as well and use the function directly before I merge the PR? Please re-request a review afterwards.

**Review by thomaspaulmann — CHANGES_REQUESTED** [review]

File: extensions/spotify-controls/package.json [hunk 4]
```diff
@@ -39,6 +39,20 @@
       "subtitle": "Spotify",
       "description": "Skips to the next track in Spotify.",
       "mode": "no-view"
+    },
+    {
+      "name": "current",
+      "title": "Current Song",
```
Comment (thomaspaulmann) [inline extensions/spotify-controls/package.json:45]:
```suggestion
      "title": "Current Track",
```

File: extensions/spotify-controls/package.json [hunk 5]
```diff
@@ -39,6 +39,20 @@
...
+      "subtitle": "Spotify",
+      "description": "Currently playing song in Spotify.",
```
Comment (thomaspaulmann) [inline extensions/spotify-controls/package.json:47]:
```suggestion
      "description": "Currently playing track in Spotify.",
```

File: extensions/spotify-controls/package.json [hunk 6]
```diff
@@ -39,6 +39,20 @@
...
+    {
+      "name": "current",
+      "title": "Current Track",
+      "subtitle": "Spotify",
+      "description": "Currently playing song in Spotify.",
+      "mode": "no-view"
+    },
+    {
+      "name": "copyUrl",
+      "title": "Copy Current Song's URL",
```
Comment (thomaspaulmann) [inline extensions/spotify-controls/package.json:52]:
```suggestion
      "title": "Copy Current Track URL",
```

File: extensions/spotify-controls/package.json [hunk 7]
```diff
@@ -39,6 +39,20 @@
...
+      "title": "Current Track",
+      "subtitle": "Spotify",
+      "description": "Currently playing track in Spotify.",
+      "mode": "no-view"
+    },
+    {
+      "name": "copyUrl",
+      "title": "Copy Current Song's URL",
+      "subtitle": "Spotify",
+      "description": "Currently playing song's URL in Spotify.",
```
Comment (thomaspaulmann) [inline extensions/spotify-controls/package.json:54]:
```suggestion
      "description": "Currently playing track's URL in Spotify.",
```

**Pull request merged**